quinine-core/src/main/scala/org/bdgenomics/quinine/rdd/RightOuterJoin.scala

```diff
@@ -0,0 +1,124 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.quinine.rdd
+
+import org.apache.spark.SparkContext._
+import org.apache.spark.rdd.RDD
+import org.bdgenomics.adam.models.{ MultiContigNonoverlappingRegions, ReferenceRegion }
+import scala.reflect.ClassTag
+
+private[quinine] object RightOuterJoin extends Serializable {
+
+  /**
+   * Performs a right outer region join.
+   *
+   * @param baseRDD Right RDD.
+   * @param joinedRDD Right RDD.
+   * @return Returns the right outer region join of two RDDs. If no value in the
+   *   left RDD overlaps with a value in the right RDD, the right RDD will be
+   *   keyed with a None. If there is an overlap, we will return key-value pairs
+   *   with all joined pairs.
+   *
+   * @tparam T The type of the value in the left RDD.
+   * @tparam U The type of the value in the right RDD.
+   */
+  def apply[T, U](
+    baseRDD: RDD[(ReferenceRegion, T)],
+    joinedRDD: RDD[(ReferenceRegion, U)])(implicit tManifest: ClassTag[T],
+                                          uManifest: ClassTag[U]): RDD[(Option[T], U)] = {
+
+    val sc = baseRDD.context
+
+    /**
+     * Original Join Design:
```

Comment:
Just out of curiosity -- my memory is failing me -- is this adapted from the code that Carl and I wrote, way back when?